tests/unit/utils/test_event_matcher.py

```diff
@@ -68,9 +69,19 @@ def test_matches_event_non_matching_pattern():
     assert not matches_event(non_matching_pattern, EVENT_DICT)
 
 
-def test_matches_event_invalid_json():
+@pytest.mark.parametrize("engine", ("python", "java"))
+def test_matches_event_invalid_json(event_rule_engine, engine):
     """Test with invalid JSON strings"""
-    with pytest.raises(json.JSONDecodeError):
+    event_rule_engine(engine)
+
+    if engine == "java":
+        # this lets the exception bubble up to the provider, when AWS returns a proper exception, it should be fixed
+        exception_type = json.JSONDecodeError
+        pytest.skip("jpype conflict")
```

Comment:
Unfortunately, we cannot risk re-introducing test coverage for the Java engine because it could break StepFunction tests using the new jsonata feature. Can we clarify that implication in a comment here because it's quite implicit knowledge and if anyone doesn't know it, re-enabling these tests risks 💥

Reply:
But this is only in the unit tests, right? I'll add the re-enabling lower down with a better comment, to be safe.

Reply:
I've moved the `event_rule_engine(engine)` after the skip to avoid enabling it